xbmc/cores/RetroPlayer/rendering/RenderUtils.cpp

```diff
@@ -55,31 +68,143 @@ void CRenderUtils::CalculateStretchMode(STRETCHMODE stretchMode,
     }
     case STRETCHMODE::Stretch4x3:
     {
-      // Stretch to 4:3 ratio
-      pixelRatio = (4.0f / 3.0f) / frameAspectRatio;
+      switch (rotationDegCCW)
+      {
+        case 90:
+        case 270:
+        {
+          pixelRatio = 1.0f / ((4.0f / 3.0f) * sourceFrameRatio);
+          break;
+        }
+        default:
+          pixelRatio = (4.0f / 3.0f) / sourceFrameRatio;
+          break;
+      }
+      zoomAmount = 1.0f;
+
+      break;
+    }
+    case STRETCHMODE::Stretch16x9:
+    {
+      switch (rotationDegCCW)
+      {
+        case 90:
+        case 270:
+        {
+          pixelRatio = 1.0f / ((16.0f / 9.0f) * sourceFrameRatio);
+          break;
+        }
+        default:
+          pixelRatio = (16.0f / 9.0f) / sourceFrameRatio;
+          break;
+      }
       zoomAmount = 1.0f;
 
       break;
     }
     case STRETCHMODE::Fullscreen:
     {
       // Stretch to the limits of the screen
-      pixelRatio = (screenWidth / screenHeight) / frameAspectRatio;
+      pixelRatio = (screenWidth / screenHeight) / sourceFrameRatio;
       zoomAmount = 1.0f;
 
       break;
     }
+    case STRETCHMODE::Integer:
+    {
+      switch (rotationDegCCW)
+      {
+        case 90:
+        case 270:
+        {
+          pixelRatio = 1.0f / (pixelRatio * sourceFrameRatio * sourceFrameRatio);
+
+          if (screenWidth / (sourceFrameRatio * pixelRatio) > screenHeight)
+          {
+            int zoomFactor = (int)screenHeight / sourceWidth;
```

Comment:
Please use c++ casts, e.g. static_cast<int>(screenHeight / sourceWidth)

Also, slap a `const` on the var, no need for it to be mutable.